test/performance/io/format_fasta_benchmark.cpp

```diff
@@ -0,0 +1,202 @@
+        }
+    }
+
+    BENCHMARK(write2);
+#endif
+
+static void read3(benchmark::State& state)
+{
+    std::string dummy_file{};
+    for (size_t idx = 0; idx < 10000000; idx++)
+        dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+    std::stringstream istream{dummy_file};
+    sequence_file_format_fasta format;
+    sequence_file_input_options<dna5, false> options;
+    std::string id;
+    dna5_vector seq;
+    for (auto _ : state)
+    {
+        format.read(istream, options, seq, id, std::ignore);
+    }
+}
+BENCHMARK(read3);
+
+static void read3_from_file(benchmark::State& state)
+{
+    // Create fasta file
+    std::string filename{"Test_Format_Fasta_Benchmark.fasta"};
+    sequence_file_output fout{filename};
+    for (size_t idx = 0; idx < 10000000; idx++)
+    {
+        std::string id{ "seq" };
+        dna5_vector seq{ "ACTAGACTAGCTACGATCAGCTACGATCAGCTACGA"_dna5 };
+        fout.push_back(std::tie(seq, id));   // as a tuple
+    }
+
+    sequence_file_input fin{filename};
+    auto file_it = fin.begin();
+    std::string id;
+    dna5_vector seq;
+
+    for (auto _ : state)
+    {
+        auto record = *file_it;
+        ++file_it;
+    }
+}
+
+BENCHMARK(read3_from_file);
+
+
+#if __has_include(<seqan/seq_io.h>)
+
+#include <fstream>
+
+std::fstream* createFastAFile(seqan::CharString &tempFilename)
+{
+    using namespace seqan;
+
+    tempFilename = SEQAN_TEMP_FILENAME();
+    char filenameBuffer[1000];
+    strncpy(filenameBuffer, toCString(tempFilename), 999);
+
+    std::fstream *file = new std::fstream(filenameBuffer, std::ios_base::in | std::ios_base::out | std::ios_base::binary | std::ios_base::trunc);
+    SEQAN_ASSERT(file->is_open());
+
+    std::string dummy_file{};
+    for (size_t idx = 0; idx < 10000000; idx++)
+        dummy_file += ">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n";
+
+    char const * STR = dummy_file.c_str();
+
+    file->write(STR, strlen(STR));
+    file->seekg(0);
+    file->seekp(0);
+    return file;
+}
+
+//TODO: How to read in seqan2 from stringstream?
+/*    static void read2(benchmark::State& state)
+    {
+        seqan::CharString meta;
+        seqan::Dna5String seq;
+        std::istringstream stream{">seq\nACTAGACTAGCTACGATCAGCTACGATCAGCTACGA\n"};
+
+        auto stream_view = view::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                          decltype(std::istreambuf_iterator<char>{})>
+                            {std::istreambuf_iterator<char>{stream},
+                             std::istreambuf_iterator<char>{}};
+
+        for (auto _ : state)
+        {
+            seqan::readRecord(meta,seq, begin(stream_view), seqan::Fasta());
+        }
+
+    }
+    BENCHMARK(read2);
+*/
+    static void read2_from_file(benchmark::State& state)
+   {
+       seqan::CharString filename;
+       std::fstream *file = createFastAFile(filename);
+       seqan::SeqFileIn reader(*file);
```

Comment:
I actually think you can just use the file you already have written down:
```suggestion
       seqan::SeqFileIn reader("Test_Format_Fasta_Benchmark.fasta");
```